packages/interface/src/peer-id/index.ts

```diff
@@ -4,39 +4,73 @@ import type { MultihashDigest } from 'multiformats/hashes/interface'
 
 export type PeerIdType = KeyType | string
 
-export interface RSAPeerId extends PeerId {
+export interface RSAPeerId {
   readonly type: 'RSA'
   readonly publicKey?: Uint8Array
+  readonly multihash: MultihashDigest
+
+  toString(): string
+  toCID(): CID
+  toBytes(): Uint8Array
+  equals(other?: PeerId | Uint8Array | string): boolean
 }
 
-export interface Ed25519PeerId extends PeerId {
+export interface Ed25519PeerId {
   readonly type: 'Ed25519'
   readonly publicKey: Uint8Array
+  readonly multihash: MultihashDigest
+
+  toString(): string
+  toCID(): CID
+  toBytes(): Uint8Array
+  equals(other?: PeerId | Uint8Array | string): boolean
 }
 
-export interface Secp256k1PeerId extends PeerId {
+export interface Secp256k1PeerId {
   readonly type: 'secp256k1'
   readonly publicKey: Uint8Array
-}
+  readonly multihash: MultihashDigest
 
-export interface URLPeerId extends PeerId {
-  readonly type: 'url'
+  toString(): string
+  toCID(): CID
+  toBytes(): Uint8Array
+  equals(other?: PeerId | Uint8Array | string): boolean
 }
 
-export interface PeerId {
-  type: PeerIdType
-  multihash: MultihashDigest
-  privateKey?: Uint8Array
-  publicKey?: Uint8Array
```

Comment:
Should we take this opportunity to do something that requires a little less maintenance for these very common interfaces?

```ts

export interface LocalPeerIdBase {
  readonly publicKey: Uint8Array
  readonly privateKey: Uint8Array
}

export interface PeerIdBase <pType extends PeerIdType, key extends Uint8Array = Uint8Array> {
  readonly type: pType
  readonly publicKey: key
  readonly multihash: MultihashDigest

  toString(): string
  toCID(): CID
  toBytes(): Uint8Array
  equals(other?: PeerId | Uint8Array | string): boolean
}

export type RSAPeerId = PeerIdBase<'RSA'>
export type LocalRSAPeerId = RSAPeerId & LocalPeerIdBase
export type PeerId = RSAPeerId | Ed25519PeerId | Secp256k1PeerId | URLPeerId
```

[basic demo on typescript playground](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgAoWgSQCYBUCeADigN4C+AUKJLIigLICuANmMABZwDO7AIsAHMIXMMggAPSCGxdkAVWoAOAIJQocfMnJVw0eEjQYoOLZWp66yAMKZeYyRGmyAymCigBpihQmEA9lCi5rQGADJ+CHDM6FjYAELcpBTIyFAQcNh+IMyahIwARszACADSEPgAXPJKqur4yanpmdm57gBucJBlldXgKmoaFJS+AUG6ISgxxvGJyAA8hATE9lIyhrFLEAA0yADW5SuOawp9tRrIALy9YP11AHxaDWkZWTnIYEQQVYufT02vuQKRVK5Sq+3qKWezTeAFsWGxODwqkxWBxuHxBMIwN4UmA-K53CABAAKACUVREhIEDTxNl4ZKqdJpfji+EgXAZ11ugxSEAAjowohy-GB2NAAPxVKYmAA+XLOmjllI85OQ+T8fmY6RAQ28I0C70+yAASs5lNLsJd1tMElwIHMAOSm5QOu4+cT+A0fZbhSLMZ0Wq0BowmABkyF9UQttog3gA9HHkAgsiJkDANVVg7ErSQKMmQKn034qpH-WbA1dc7jPlUnWaHVsGnlCsVulUQBAAO7ygb4YkAbQAuqTGylCO1OhA28gO92TjcFQPh6PkHDUYj2O2uz26kuR8yCR4yVpGmBGFAQMgHeywA7kGQV7TbMeSKfz5fr1i7w-maz2S+3wvGdt3nbk+1Je8V35QVmGFUVoAg180jPID4FglAf0ofMuE1CAADpmD8Eki1JIA)

Reply:
idk that we're planning on adding many more peerId types or anything, but it results in less code that seems easier to reason about for me at least. and will be less lines in the future if we need to extend PeerIdBase

Reply:
I somewhat did this on purpose.  As suggested the generated API docs will have you clicking through from `PeerId` ->  `Ed25519PeerId` -> `PeerIdBase` to see the docs for the fields on the type.

No-one will ever deal with a `PeerIdBase` in day to day code so I just copied the fields/methods to make the docs simpler.